src/extension-support/extension-manager.js

```diff
@@ -391,23 +391,22 @@ class ExtensionManager {
             }
             break;
         default:
-            if (!blockInfo.opcode) {
-                throw new Error('Missing opcode for block');
-            }
-
-            blockInfo.func = blockInfo.func ? this._sanitizeID(blockInfo.func) : blockInfo.opcode;
+            if (blockInfo.opcode) {
+                const funcName = blockInfo.func ? this._sanitizeID(blockInfo.func) : blockInfo.opcode;
 
-            // Avoid promise overhead if possible
-            if (dispatch._isRemoteService(serviceName)) {
-                blockInfo.func = dispatch.call.bind(dispatch, serviceName, blockInfo.func);
-            } else {
-                const serviceObject = dispatch.services[serviceName];
-                const func = serviceObject[blockInfo.func];
-                if (func) {
-                    blockInfo.func = func.bind(serviceObject);
+                // Avoid promise latency unless necessary
+                if (dispatch._isRemoteService(serviceName)) {
+                    blockInfo.func = (args, util) => dispatch.call(serviceName, funcName, args, util, blockInfo);
                 } else {
-                    throw new Error(`Could not find extension block function called ${blockInfo.func}`);
+                    const serviceObject = dispatch.services[serviceName];
+                    if (!serviceObject[funcName]) {
+                        // The function might show up later as a dynamic property of the service object
+                        log.warn(`Could not find extension block function called ${funcName}`);
```

Comment:
Since it's looking up `serviceObject[funcName]` in the body of the (anonymous) function, it's possible that `serviceObject[funcName]` will become a valid function call between now and the first time it gets called. This could happen if the extension dynamically generates methods and monkey-patches itself, for example, especially if it needs to download some data first.

I don't expect this to be common but technically the way we make "remote" service calls doesn't even do this check at all, so I figured that if it's allowed there it should be allowed here. 🤷‍♂️